accounts/abi/bind/backends/blockchain.go

```diff
@@ -141,3 +144,197 @@ func (b *BlockchainContractCaller) getBlockAndState(num *big.Int) (*types.Block,
 	state, err := b.bc.StateAt(block.Root())
 	return block, state, err
 }
+
+type BlockchainContractTransactor struct {
+	bc     BlockChainForCaller
+	txPool *blockchain.TxPool
+}
+
+// This nil assignment ensures at compile time that BlockchainContractCaller implements bind.ContractCaller.
+var _ bind.ContractTransactor = (*BlockchainContractTransactor)(nil)
+
+func NewBlockchainContractTransactor(bc BlockChainForCaller, txPool *blockchain.TxPool) *BlockchainContractTransactor {
+	return &BlockchainContractTransactor{bc, txPool}
+}
+
+func (b *BlockchainContractTransactor) PendingCodeAt(ctx context.Context, account common.Address) ([]byte, error) {
+	// TODO-Klaytn this is not pending code but latest code
+	state, err := b.bc.State()
+	if err != nil {
+		return nil, err
+	}
+	return state.GetCode(account), nil
+}
+
+func (b *BlockchainContractTransactor) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error) {
+	if b.txPool != nil {
```

Comment:
Under what conditions can `b.txPool` be nil? If the NewBlockchainContractTransactor already includes checks for both bc and txPool, it might render these checks unnecessary. Therefore, a single statement like return b.txPool.GetPendingNonce(account), nil could suffice in my view.

Reply:
`txPool` should be provided from the caller of the `BlockchainContractTransactor` instance. If the caller didn't provide `txPool` for the corresponding blockchain backend (e.g., pass `nil` in the argument), the `txPool` would be `nil`.

I could force caller to pass non-nil argument for `txPool`, however there are some functions that does not depend on the `txPool` in `BlockchainContractTransactor`. So instead I added above checks in functions that uses `txPool`.

Reply:
Relocation the check to the inside of `NewBlockchainContractBackend` does not clearly solve?

Reply:
Yes, that could be one option.

I thought it is OK to allow calller to pass `nil` for `txPool` when calling `NewBlockchainContractTransactor`. Then, the caller could only use methods that does not depend on `txPool` (e.g., `EstimateGas`). I'm not sure how many cases would use `BlockchainContractTransactor` like that way.

I'll update to force `txPool` to be non-nil.

Reply:
If we unify the classes into one BlockchainContractBackend (see https://github.com/klaytn/klaytn/pull/1944#discussion_r1308460561), `b.txPool` may be nil as @2dvorak explained.

I'd say we keep the nil check.